src/control/server/mgmt_check.go

```diff
@@ -511,6 +511,24 @@ func (svc *mgmtSvc) SystemCheckSetPolicy(ctx context.Context, req *mgmtpb.CheckS
 		return nil, err
 	}
 
+	// 'check set-policy' only changes policy, not flags, then reuse 'Flags' to indicate
+	// something internally, such as no dRPC call for test purpose.
+	if req.Flags == 0 {
```

Comment:
We have ways of mocking a dRPC call in tests, so this shouldn't be necessary. Better to test the production code as much as possible here. I can help with this if you would like, and there are [some examples using a mockDrpcClient](https://github.com/daos-stack/daos/blob/748be8f6b3e4a5c6f4517a51ec23c2d4f23fb46f/src/control/server/mgmt_check_test.go#L172) in our test code.

Reply:
Thanks @kjacque . I will make your PR as another commit under my current PR, then they can be tested and landed together.